clients/drcachesim/tests/raw2trace_unit_tests.cpp

```diff
@@ -3156,6 +3156,248 @@ test_ifiltered(void *drcontext)
+        instr_t *move =
+            XINST_CREATE_move(drcontext, opnd_create_reg(REG2), opnd_create_reg(REG1));
+        instrlist_append(ilist, nop);
+        instrlist_append(ilist, load);
+        instrlist_append(ilist, store);
+        instrlist_append(ilist, move);
+        size_t offs_nop = 0;
+        size_t offs_load = offs_nop + instr_length(drcontext, nop);
+        size_t offs_store = offs_load + instr_length(drcontext, load);
+        size_t offs_move = offs_store + instr_length(drcontext, store);
+
+        std::vector<offline_entry_t> raw;
+        raw.push_back(make_header());
+        raw.push_back(make_tid());
+        raw.push_back(make_pid());
+        raw.push_back(make_line_size());
+        raw.push_back(make_timestamp());
+        raw.push_back(make_core());
+        raw.push_back(make_block(offs_load, 3));
+        raw.push_back(make_memref(42));
+        raw.push_back(make_marker(TRACE_MARKER_TYPE_KERNEL_EVENT, offs_store));
+        raw.push_back(make_block(offs_move, 1));
+        raw.push_back(make_exit());
+
+        std::vector<uint64_t> stats;
+        std::vector<trace_entry_t> entries;
+        if (!run_raw2trace(drcontext, raw, ilist, entries, &stats))
+            return false;
+        int idx = 0;
+        res &=
+            (check_entry(entries, idx, TRACE_TYPE_HEADER, -1) &&
+             check_entry(entries, idx, TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_VERSION) &&
+             check_entry(entries, idx, TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_FILETYPE) &&
+             check_entry(entries, idx, TRACE_TYPE_THREAD, -1) &&
+             check_entry(entries, idx, TRACE_TYPE_PID, -1) &&
+             check_entry(entries, idx, TRACE_TYPE_MARKER,
+                         TRACE_MARKER_TYPE_CACHE_LINE_SIZE) &&
+             check_entry(entries, idx, TRACE_TYPE_MARKER,
+                         TRACE_MARKER_TYPE_CHUNK_INSTR_COUNT) &&
+             check_entry(entries, idx, TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_TIMESTAMP) &&
+             check_entry(entries, idx, TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_CPU_ID) &&
+             // The load instruction.
+             check_entry(entries, idx, TRACE_TYPE_ENCODING, -1) &&
+#ifdef X86_32
+             // An extra encoding entry is needed.
+             check_entry(entries, idx, TRACE_TYPE_ENCODING, -1) &&
+#endif
+             check_entry(entries, idx, TRACE_TYPE_INSTR, -1, offs_load) &&
+             check_entry(entries, idx, TRACE_TYPE_READ, -1) &&
+             // The store instruction and the memref are removed.
+             check_entry(entries, idx, TRACE_TYPE_MARKER,
+                         TRACE_MARKER_TYPE_UNCOMPLETED_INSTRUCTION) &&
+             check_entry(entries, idx, TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_KERNEL_EVENT,
+                         offs_store) &&
+             check_entry(entries, idx, TRACE_TYPE_ENCODING, -1) &&
+             check_entry(entries, idx, TRACE_TYPE_INSTR, -1, offs_move) &&
```

Comment:
Wait: why is the move here?  Wouldn't we either go to a signal handler or exit if no handler?  Why is this different from the prior case?